lumeer-core/src/main/java/io/lumeer/core/util/coordinates/CoordinatesParser.java

```diff
@@ -0,0 +1,251 @@
+    *
+    * Supported standard formats are the following, with dots or optionally a comma as the decimal marker, and variations
+    * on the units also accepted e.g. °, d, º, g, o.
+    * <ul>
+    *   <li>43.63871944444445</li>
+    *   <li>N43°38'19.39"</li>
+    *   <li>43°38'19.39"N</li>
+    *   <li>43°38.3232'N</li>
+    *   <li>43d 38m 19.39s N</li>
+    *   <li>43 38 19.39</li>
+    *   <li>433819N</li>
+    * </ul>
+    *
+    * @param latitude  The decimal latitude
+    * @param longitude The decimal longitude
+    *
+    * @return The parse result
+    */
+   public static Optional<LatLng> parseLatLng(final String latitude, final String longitude) {
+      if (latitude == null || "".equals(latitude) || longitude == null || "".equals(longitude)) {
+         return Optional.empty();
+      }
+      Double lat = NumberParser.parseDouble(latitude);
+      Double lng = NumberParser.parseDouble(longitude);
+      if (lat == null || lng == null) {
+         // try degree minute seconds
+         try {
+            lat = parseDMS(latitude, true);
+            lng = parseDMS(longitude, false);
+         } catch (IllegalArgumentException e) {
+            return Optional.empty();
+         }
+      }
+
+      return validateAndRound(lat, lng);
+   }
+
+   // 02° 49' 52" N	131° 47' 03" E
+   public static Optional<LatLng> parseVerbatimCoordinates(final String coordinates) {
+      if (coordinates == null || "".equals(coordinates)) {
+         return Optional.empty();
+      }
+      Matcher m = DMS_COORD.matcher(coordinates);
+      if (m.find()) {
+         final String dir1 = m.group(4);
+         final String dir2 = m.group(8);
+         // first parse coords regardless whether they are lat or lon
+         double c1 = coordFromMatcher(m, 1,2,3, dir1);
+         double c2 = coordFromMatcher(m, 5,6,7, dir2);
+         // now see what order the coords are in:
+         if (isLat(dir1) && !isLat(dir2)) {
+            return validateAndRound(c1, c2);
+
+         } else if (!isLat(dir1) && isLat(dir2)) {
+            return validateAndRound(c2, c1);
+
+         } else {
+            return Optional.empty();
+         }
+
+      } else if(coordinates.length() > 4) {
+         // try to split and then use lat/lon parsing
+         for (final char delim : ",;/ ".toCharArray()) {
+            int cnt = StringUtils.countMatches(coordinates, String.valueOf(delim));
+            if (cnt == 1) {
+               String[] latlon = StringUtils.split(coordinates, delim);
+               if (latlon.length == 2) {
+                  return parseLatLng(latlon[0], latlon[1]);
+               }
+            }
+         }
+      }
+      return Optional.empty();
+   }
+
+   private static Optional<LatLng> validateAndRound(double lat, double lon) {
+      // round to 6 decimals
+      final double latOrig = lat;
+      final double lngOrig = lon;
+      lat = roundTo6decimals(lat);
+      lon = roundTo6decimals(lon);
```

Comment:
Why are we rounding it here to 6 digits? Should not user be able to specify the precision?

Reply:
The sixth decimal place is worth up to 0.11 m: you can use this for laying out structures in detail, for designing landscapes, building roads. It should be more than good enough for tracking movements of glaciers and rivers. This can be achieved by taking painstaking measures with GPS, such as differentially corrected GPS.
The seventh decimal place is worth up to 11 mm: this is good for much surveying and is near the limit of what GPS-based techniques can achieve.
The eighth decimal place is worth up to 1.1 mm: this is good for charting motions of tectonic plates and movements of volcanoes. Permanent, corrected, constantly-running GPS base stations might be able to achieve this level of accuracy.